src/app/_components/Event/EventSection.js

```diff
@@ -0,0 +1,154 @@
+
+  const searchParams = useSearchParams();
+  const { add: addParam, remove: removeParam } = useSetParam('month');
+
+  const { ref, inView } = useInView();
+
+  const { semiorange } = buttonColorVariant.eventFilter;
+
+  const activeButtonStyles = cn({
+    'pointer-events-none border-secondary-300 bg-secondary-300 font-semibold text-gray-900': semiorange.active,
+  });
+
+  function selectButtonIcon(index) {
+    if (activeMonth - 1 === index) {
+      return (
+        <CheckMark
+          key={`checkmark+${index}`}
+          className={cn(`block h-4 w-4 transition-all group-hover:hidden group-focus:hidden`)}
+        />
+      );
+    }
+
+    return (
+      <Search
+        key={`searchicon+${index}`}
+        className="hidden h-4 w-4 transition-all group-hover:block group-focus:block"
+      />
+    );
+  }
+
+  // Fetch data and pass url params
+  const { data, error, isLoading, hasNextPage, fetchNextPage, isSuccess, isFetchingNextPage } = useInfiniteQuery({
+    queryFn: ({ month = activeMonth, pageParam = '' }) => allEvents({ month, take: 6, lastCursor: pageParam }),
+    queryKey: ['event', activeMonth],
+    getNextPageParam: lastPage => lastPage?.metaData.lastCursor,
+  });
+
+  // Get data on first render or on url open
+  useEffect(() => {
+    const monthFromQuery = searchParams.get('month');
+
+    if (monthFromQuery && monthFromQuery !== currentMonth) {
+      // If the month is defined in the query and is not equal to the current month
+      setActiveMonth(monthFromQuery);
+      removeParam();
+      addParam(monthFromQuery);
+    } else if (!monthFromQuery || monthFromQuery === undefined) {
+      setActiveMonth(currentMonth);
+      removeParam();
+      addParam(currentMonth);
+    }
+    // eslint-disable-next-line
+  }, [currentMonth]);
+
+  useEffect(() => {
+    // if the last element is in view and there is a next page, fetch the next page
+    if (inView && hasNextPage) {
+      fetchNextPage();
+    }
+    // eslint-disable-next-line
+  }, [hasNextPage, inView]);
+
+  // Filter data based on selected month
+  const handleFilter = newMonth => {
+    setActiveMonth(newMonth);
+    removeParam();
+    addParam(newMonth.toString());
+  };
+
+  return (
+    <>
+      <div className="lg:w-max-[900px] mx-auto flex w-full flex-col items-start justify-start gap-6 self-stretch">
+        <div className="flex flex-row flex-wrap items-start justify-start gap-3">
+          {filteredMonths.map(month => (
+            <PillButton
+              variant="eventFilter"
+              colorVariant="semiorange"
+              className={cn(activeMonth - 1 === month.index && activeButtonStyles, 'group w-fit')}
+              key={month.index}
+              onClick={() => {
+                handleFilter(month.index + 1);
+              }}
+              icon={selectButtonIcon(month.index)}
+            >
+              {month.name.charAt(0).toUpperCase() + month.name.slice(1)}
```

Comment:
@Yivgha Do you have capitalize util in project? If not, it can be useful to create one

Reply:
changed